unit_tests/testConversions/testConversions.f90

```diff
@@ -31,75 +31,166 @@ SUBROUTINE testTemperatures()
-  ASSERT_EQ(F_to_K(32.0_SRK),273.15_SRK,'32 F_to_K')
-  ASSERT_EQ(F_to_K(68.0_SRK),293.15_SRK,'68 F_to_K')
-  ASSERT_EQ(F_to_K(212.0_SRK),373.15_SRK,'212 F_to_K')
-  ASSERT_EQ(F_to_K(557.33_SRK),565.0_SRK,'557.33 F_to_K')
+  ASSERT_APPROXEQ(out_temp(1),273.15_SRK,'32 F_to_K (1D)')
+  ASSERT_APPROXEQ(out_temp(2),293.15_SRK,'68 F_to_K (1D)')
+  ASSERT_APPROXEQ(out_temp(3),373.15_SRK,'212 F_to_K (1D)')
+  ASSERT_APPROXEQ(out_temp(4),565.0_SRK,'557.33 F_to_K (1D)')
+  ASSERT_APPROXEQ(F_to_K(32.0_SRK),273.15_SRK,'32 F_to_K')
+  ASSERT_APPROXEQ(F_to_K(68.0_SRK),293.15_SRK,'68 F_to_K')
+  ASSERT_APPROXEQ(F_to_K(212.0_SRK),373.15_SRK,'212 F_to_K')
+  ASSERT_APPROXEQ(F_to_K(557.33_SRK),565.0_SRK,'557.33 F_to_K')
 
   COMPONENT_TEST('K_to_F')
   in_temp = (/273.15_SRK,293.15_SRK,373.15_SRK,565.0_SRK/)
   out_temp = K_to_F(in_temp)
-  ASSERT_EQ(out_temp(1),32.0_SRK,'273.15 K_to_F (1D)')
-  ASSERT_EQ(out_temp(2),68.0_SRK,'293.15 K_to_F (1D)')
-  ASSERT_EQ(out_temp(3),212.0_SRK,'373.15 K_to_F (1D)')
-  ASSERT_EQ(out_temp(4),557.33_SRK,'565 K_to_F (1D)')
-  ASSERT_EQ(K_to_F(273.15_SRK),32.0_SRK,'273.15 K_to_F')
-  ASSERT_EQ(K_to_F(293.15_SRK),68.0_SRK,'293.15 K_to_F')
-  ASSERT_EQ(K_to_F(373.15_SRK),212.0_SRK,'373.15 K_to_F')
-  ASSERT_EQ(K_to_F(565.0_SRK),557.33_SRK,'565 K_to_F')
+  ASSERT_APPROXEQ(out_temp(1),32.0_SRK,'273.15 K_to_F (1D)')
+  ASSERT_APPROXEQ(out_temp(2),68.0_SRK,'293.15 K_to_F (1D)')
+  ASSERT_APPROXEQ(out_temp(3),212.0_SRK,'373.15 K_to_F (1D)')
+  ASSERT_APPROXEQ(out_temp(4),557.33_SRK,'565 K_to_F (1D)')
+  ASSERT_APPROXEQ(K_to_F(273.15_SRK),32.0_SRK,'273.15 K_to_F')
+  ASSERT_APPROXEQ(K_to_F(293.15_SRK),68.0_SRK,'293.15 K_to_F')
+  ASSERT_APPROXEQ(K_to_F(373.15_SRK),212.0_SRK,'373.15 K_to_F')
+  ASSERT_APPROXEQ(K_to_F(565.0_SRK),557.33_SRK,'565 K_to_F')
 
+  COMPONENT_TEST('tempTo_C')
+  !To Celsius from Kelvin
+  in_temp = (/0.0_SRK,273.15_SRK,565.0_SRK,1200.0_SRK/)
+  out_temp = tempTo_C(in_temp,'K')
+  ASSERT_APPROXEQ(out_temp(1),-273.15_SRK,'0 K_to_C (1D)')
+  ASSERT_APPROXEQ(out_temp(2),0.0_SRK,'273.15 K_to_C (1D)')
+  ASSERT_APPROXEQ(out_temp(3),291.85_SRK,'565 K_to_C (1D)')
+  ASSERT_APPROXEQ(out_temp(4),926.85_SRK,'1200 K_to_C (1D)')
+  !To Celsius from Fahrenheit
+  in_temp = (/32.0_SRK,68.0_SRK,212.0_SRK,557.33_SRK/)
+  out_temp = tempTo_C(in_temp,'F')
+  ASSERT_APPROXEQ(out_temp(1),0.0_SRK,'32 F_to_C (1D)')
+  ASSERT_APPROXEQ(out_temp(2),20.0_SRK,'68 F_to_C (1D)')
+  ASSERT_APPROXEQ(out_temp(3),100.0_SRK,'212 F_to_C (1D)')
+  ASSERT_APPROXEQ(out_temp(4),291.85_SRK,'557.33 F_to_C (1D)')
+  !To Celsius from Rankine
+  in_temp = (/0.0_SRK,273.15_SRK,546.3_SRK,1475.01_SRK/)
+  out_temp = tempTo_C(in_temp,'R')
+  ASSERT_APPROXEQF(out_temp(1),-273.15_SRK,'0 R_to_C (1D)')
+  ASSERT_APPROXEQF(out_temp(2),-121.4_SRK,'273.15 R_to_C (1D)')
+  ASSERT_APPROXEQF(out_temp(3),30.35_SRK,'565 R_to_C (1D)')
+  ASSERT_APPROXEQF(out_temp(4),546.3_SRK,'1200 R_to_C (1D)')
+
+  COMPONENT_TEST('C_to_K')
```

Comment:
Is this different from the above "C_to_K" component test?

Reply:
Yeah, that should have changed to "tempTo_K"